src/data-server/sbvr-server.js

```diff
@@ -76,39 +76,40 @@ ADD COLUMN IF NOT EXISTS "modified at" TIMESTAMP DEFAULT CURRENT_TIMESTAMP NOT N
 export function setup(app, sbvrUtils, db) {
 	const uiApi = sbvrUtils.api.ui;
 	const devApi = sbvrUtils.api.dev;
-	const setupModels = (
+	const setupModels = async (
 		/** @type { import('../database-layer/db').Tx } */ tx,
 	) => {
-		const uiApiTx = uiApi.clone({
-			passthrough: {
-				tx,
-				req: permissions.root,
-			},
-		});
-		return uiApiTx
-			.get({
-				resource: 'textarea',
-				options: {
-					$select: 'id',
-					$filter: {
-						name: 'model_area',
-					},
+		try {
+			const uiApiTx = uiApi.clone({
+				passthrough: {
+					tx,
+					req: permissions.root,
 				},
-			})
-			.then((/** @type { Array<{ [key: string]: any }> } */ result) => {
-				if (result.length === 0) {
-					// Add a model_area entry if it doesn't already exist.
-					return uiApiTx.post({
-						resource: 'textarea',
-						body: {
+			});
+			await uiApiTx
+				.get({
+					resource: 'textarea',
+					options: {
+						$select: 'id',
+						$filter: {
 							name: 'model_area',
-							text: ' ',
 						},
-					});
-				}
-			})
-			.then(() =>
-				devApi.get({
+					},
+				})
+				.then((/** @type { Array<{ [key: string]: any }> } */ result) => {
```

Comment:
Missed that `.then` on purpose?
I guess it's the type that made it harder.

Reply:
Yeah, I can't force the typing if I convert it to async/await (it requires specific brackets that prettier would remove), so it can stay this way until the file is in typescript